### PR TITLE
Renamed package internal/log to internal/debug

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
In preparation for other debug things.

### CODE REVIEW

File: internal/debug/debug_debug.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
+//go:build debug
```
Comment (richardpark-msft) [inline internal/debug/debug_debug.go:1]: This file name is awesome.

File: internal/debug/debug_debug.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		return
+	}
+
+	debugLevel = level
+}
+
+// Log writes the formatted string to stderr.
+// Level indicates the verbosity of the messages to log.
+// The greater the value, the more verbose messages will be logged.
+func Log(level int, format string, v ...interface{}) {
```
Comment (richardpark-msft) [inline internal/debug/debug_debug.go:30]: Doesn't have to be done now, but it'd be good to document the expected range. I think we use 1 and 3, so maybe those should get constants and names.

**Review by richardpark-msft — APPROVED** [review]
Looks good, this will definitely help with testing.

File: internal/debug/debug_debug.go (hunk 2 above)
Comment (jhendrixMSFT) [reply internal/debug/debug_debug.go:30]: My preference would be to remove it entirely as I don't think it provides much value.  I think we can do better with our logs in general, I opened https://github.com/Azure/go-amqp/issues/187 to track this.

**Pull request merged**